pyresample/future/geometry/area.py

```diff
@@ -19,4 +19,105 @@
+        attrs:
+            Arbitrary metadata related to the area. Some keys in this
+            dictionary have special meaning and may be used for various
+            logging or serialization processes. The primary special keys are:
+
+            * name: The identifying name for this area. This is equivalent
+                to 'area_id' in the legacy AreaDefinition class. It is used
+                in YAML serialization if provided. Defaults to an empty string
+                if not provided. Not providing this should not affect normal
+                coordinate operations.
+            * description: A human-readable description of the area. This is
+                equivalent to 'description' in the legacy AreaDefinition class.
+                It is used in YAML serialization if provided.
+
+    Attributes:
+        area_id (str):
+            Identifier for the area. This is a convenience for backwards
+            compatibility and accesses the ``.attrs['name']`` metadata.
+            This will be set to an empty string if not provided.
+        width (int):
+            x dimension in number of pixels, aka number of grid columns
+        height (int):
+            y dimension in number of pixels, aka number of grid rows
+        size (int):
+            Number of points in grid
+        area_extent_ll (tuple):
+            Area extent in lons lats as a tuple (lower_left_lon, lower_left_lat, upper_right_lon, upper_right_lat)
+        pixel_size_x (float):
+            Pixel width in projection units
+        pixel_size_y (float):
+            Pixel height in projection units
+        upper_left_extent (tuple):
+            Coordinates (x, y) of upper left corner of upper left pixel in projection units
+        pixel_upper_left (tuple):
+            Coordinates (x, y) of center of upper left pixel in projection units
+        pixel_offset_x (float):
+            x offset between projection center and upper left corner of upper
+            left pixel in units of pixels.
+        pixel_offset_y (float):
+            y offset between projection center and upper left corner of upper
+            left pixel in units of pixels.
+        crs (CRS):
+            Coordinate reference system object similar to the PROJ parameters in
+            `proj_dict` and `proj_str`. This is the preferred attribute to use
+            when working with the `pyproj` library. Note, however, that this
+            object is not thread-safe and should not be passed between threads.
+        attrs (dict):
+            Arbitrary metadata related to the area.
+
+    """
+
+    def __init__(
+            self,
+            crs: Union[str, int, dict, CRS],
+            width: int,
+            height: int,
+            area_extent: tuple[float, float, float, float],
+            attrs: Optional[dict] = None
+    ):
+        # FUTURE: Maybe do `shape` instead of the separate height and width
```

Comment:
yes, that would allow to have a vertical/z dimension too.

Reply:
Ok, I'll make a PR for this. This would likely require some of the "feature flag" stuff I talked about in my other comments.